scripts/end_to_end_test/run_end_to_end_test.py

```diff
@@ -610,6 +686,30 @@ def run_steps(test_yaml, test_instance):
 
 
 def run_test(build_directory, yaml_file, node_exe, name_filter=None):
+    global command_map
+
+    command_map = {
```

Comment:
We never write to `command_map`, so I would make it a constant at file scope (renamed to COMMAND_MAP for style). That way you would not need all the `global` declarations either.